src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/DoNotCompareSyntaxTokenAgainstDefaultLiteralFix.cs

```diff
@@ -0,0 +1,95 @@
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers
+{
+    /// <summary>
+    /// RS1034: Prefer token.IsKind(SyntaxKind.None) or token.RawKind == 0 over token == default.
+    /// </summary>
+    internal abstract class DoNotCompareSyntaxTokenAgainstDefaultLiteralFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DiagnosticIds.DoNotCompareSyntaxTokenAgainstDefaultLiteralRuleId);
+
+        protected abstract ISyntaxFacts SyntaxFacts { get; }
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        CodeAnalysisDiagnosticsResources.DoNotCompareSyntaxTokenAgainstDefaultLiteralCodeFix,
+                        cancellationToken => FixAsync(context.Document, diagnostic.Location.SourceSpan, cancellationToken),
+                        equivalenceKey: nameof(DoNotCompareSyntaxTokenAgainstDefaultLiteralFix)),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private async Task<Document> FixAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var comparisonNode = root.FindNode(span, getInnermostNodeForTie: true);
+
+            if (!SyntaxFacts.IsBinaryExpression(comparisonNode))
+            {
+                Debug.Fail("BinaryExpressionSyntax was expected in fixer.");
+                return document;
+            }
+
+            SyntaxFacts.GetPartsOfBinaryExpression(comparisonNode, out var left, out _, out var right);
+
+            SyntaxNode? tokenNode = null;
+            if (SyntaxFacts.IsDefaultLiteralExpression(left))
+            {
+                tokenNode = right;
+            }
+            else if (SyntaxFacts.IsDefaultLiteralExpression(right))
+            {
+                tokenNode = left;
+            }
+            else
+            {
+                Debug.Fail("A 'default' literal expression was expected in fixer.");
+                return document;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            if (semanticModel.GetMemberGroup(tokenNode, cancellationToken).Any(s => s.Name == DiagnosticWellKnownNames.IsKindName))
+            {
+                // Offer token.IsKind(SyntaxKind.None)
+                // TODO: Negate if the original comparison is !=.
```

Comment:
There are multiple ways to know what the operator is, I'm not sure which way you'd like?

- Using additionalProperties from the analyzer.
- Adding the operators kinds to ISyntaxKinds.
- Or is there a better option?